apps/leo_storage/src/leo_storage_handler_object.erl

```diff
@@ -823,6 +903,40 @@ replicate(DestNodes, AddrId, Key) ->
     end.
 
 
+%% @private
+-spec(get_inconsistent_nodes(AddrId, Preferred_N, InconsistentNodes) ->
+             [Node] | {error, Cause} when AddrId::integer(),
+                                          Preferred_N::non_neg_integer(),
+                                          Node::node(),
+                                          InconsistentNodes::[Node],
+                                          Cause::any()).
+get_inconsistent_nodes(AddrId, Preferred_N, InconsistentNodes) ->
+    case leo_redundant_manager_api:get_redundancies_by_addr_id(AddrId) of
+        {ok, #redundancies{nodes = Redundancies,
+                           n = N}} ->
+            case (N > Preferred_N andalso Preferred_N > 0) of
+                true ->
+                    lists:reverse(
+                      lists:foldl(
+                        fun(Node, Acc) ->
+                                case lists:filter(
+                                       fun(#redundant_node{node = RedundantNode}) ->
+                                               Node == RedundantNode
+                                       end, lists:sublist(Redundancies, Preferred_N)) of
```

Comment:
Since `lists:sublist(Redundancies, Preferred_N))` is evaluated many times,
It would be good to move this expression to before calling `lists:reverse` for performance.